qubit-events.liquid

```diff
@@ -0,0 +1,200 @@
+</script>
+<!-- TODO: Update below path with the one provided to you by Qubit -->
+<script src="//static.goqubit.com/smartserve-XXXX.js" async defer></script>
+<script>
+  (function () {
+    var config = {
+      eventNamespace: false,
+    }
+
+    if(config.eventNamespace) {
+      config.ns = config.eventNamespace + "."
+    } else {
+      config.ns=""
+    }
+    config.template = {{ template | json }}
+    
+    var getPageCategory = function () {
+      switch (config.template) {
+        case "index": 
+          return "home";
+          break;
+        case "page":
+          return "content";
+          break;
+        case "article":
+          return "blog";
+          break;
+        case "product":
+          return "product";
+          break;
+        case "search":
+          return "search";
+          break;
+        case "collection": 
+          return "category";
+          break;
+        case "cart":
+          return "basket";
+          break;
+        default: 
+          return config.template;
+      }
+    };
+
+
+    {% comment %}
+		Page
+    {% endcomment %}
+    uv.emit(config.ns+'ecView', {
+      type: getPageCategory()
+    })
+
+        
+    {% comment %}
+		Product - http://docs.shopify.com/api/product
+    {% endcomment %}
+        
+    {% if product %}
+    uv.emit(config.ns+'ecProduct', {
+      eventType: 'detail',
+      product: {
+        sku: {{ product.variants.first.sku | default: product.variants.first.id | json }},
+        productId: "{{ product.id }}",
+        name: {{ product.title | json }},
+        manufacturer: {{ product.vendor | json}},
+        stock: {{ product.variants.first.inventory_quantity }},
+        price: {
+        currency: "{{ shop.currency }}",
+        value: {{ product.price | divided_by: 100 }}
+      },
+      url: "{{ product.url }}",
+      description: {{ product.description | strip_html | truncate: 100 | json }},
+      category: ["{{ product.type}}"],
+      images: {{ product.images | json }}
+    }
+            })
+
+    {% endif %}
+
+
+    {% if collection %}
+    {% comment %}
+		Listing - collection pages - http://docs.shopify.com/themes/liquid-variables/collection
+    {% endcomment %}
+    {% for product in collection.products %} 
+    uv.emit(config.ns+'ecProduct', {
+      eventType: 'listing',
+      product: {
+        sku: {{ product.variants.first.sku | json }},
+        productId: "{{ product.id }}",
+        name: {{ product.title | json }},
+        stock: {{ product.variants.first.inventory_quantity }},
+		manufacturer: {{ product.vendor | json}},
+        price: {
+        	currency: "{{ shop.currency }}",
+        	value: {{ product.price | divided_by: 100 }}
+     	 },
+      	url: "{{ product.url }}",
+     	description: {{ product.description | strip_html | truncate: 100 | json }},
```

Comment:
Why are we truncating at 100 characters? The QP `Text` type goes up to 2048 chars.